api/tests/integration/infrastructure/repositories/target-profile-attachable-for-admin-repository_test.js

```diff
@@ -0,0 +1,148 @@
+import { databaseBuilder, expect } from '../../../test-helper.js';
+import * as targetProfileAttachableForAdminRepository from '../../../../lib/infrastructure/repositories/target-profile-attachable-for-admin-repository.js';
+
+describe('Integration | Repository | target-profile-attachable-for-admin', function () {
+  describe('#findAttachable', function () {
+    it('should return target profiles ordered by name asc, then id desc', async function () {
+      // given
+      const firstResult = {
+        id: 100,
+        name: 'AAA',
+        isPublic: false,
+        outdated: false,
+      };
+      databaseBuilder.factory.buildTargetProfile(firstResult);
+
+      const secondResult = {
+        id: 300,
+        name: 'BBB',
+        isPublic: true,
+        outdated: false,
+      };
+      databaseBuilder.factory.buildTargetProfile(secondResult);
+      const thirdResult = {
+        id: 200,
+        name: 'BBB',
+        isPublic: false,
+        outdated: false,
+      };
+      databaseBuilder.factory.buildTargetProfile(thirdResult);
+
+      await databaseBuilder.commit();
+
+      // when
+      const results = await targetProfileAttachableForAdminRepository.find();
+
+      // then
+      expect(results).to.deep.equal([
+        { id: 100, name: 'AAA' },
+        { id: 300, name: 'BBB' },
+        { id: 200, name: 'BBB' },
+      ]);
+    });
+
+    it('should not return outdated target profiles', async function () {
+      // given
+      databaseBuilder.factory.buildTargetProfile({
+        id: 100,
+        name: 'NOTOUTDATED',
+        isPublic: false,
+        outdated: false,
+      });
+
+      databaseBuilder.factory.buildTargetProfile({
+        id: 300,
+        name: 'OUTDATED',
+        isPublic: true,
+        outdated: true,
+      });
+      await databaseBuilder.commit();
+
+      // when
+      const results = await targetProfileAttachableForAdminRepository.find();
+
+      // then
+      expect(results).to.deep.equal([{ id: 100, name: 'NOTOUTDATED' }]);
+    });
+
+    context('when the target profiles is not linked to any badges', function () {
+      it('should return target profiles', async function () {
```

Comment:
Suggestion à appliquer à différents endroits du fichier

```suggestion
      it('should return attachable target profiles', async function () {

Reply:
Done et changé autre part!